pkg/storage/store.go

```diff
@@ -201,3 +217,36 @@ func filterChunksByTime(from, through model.Time, chunks []chunk.Chunk) []chunk.
 	}
 	return filtered
 }
+
+func registerCustomIndexClients(cfg Config, schemaCfg chunk.SchemaConfig) {
+	boltdbShipperInstances := 0
+	storage.RegisterIndexClient(local.BoltDBShipperType, func() (chunk.IndexClient, error) {
+		// since we do not know which object client is being used for the period for which we are creating this index client,
+		// we need to iterate through all the periodic configs to find the right one.
+		// We maintain number of instances that we have already created in boltdbShipperInstances and then count the number of
+		// encounters of BoltDBShipperType until we find the right periodic config for getting the ObjectType.
+		// This is done assuming we are creating index client in the order of periodic configs.
+		// Note: We are assuming that user would never store chunks in table based store otherwise NewObjectClient would return an error.
+
+		// ToDo: Try passing on ObjectType from Cortex to the callback for creating custom index client.
+		boltdbShipperEncounter := 0
```

Comment:
Yes we do need that to be in cortex. I'm fine with having this hack in the meantime. How do we know that the callback in called in the same order of the configs ?

We need some tests too.